plugins/database/postgresql/postgresql.go

```diff
@@ -505,3 +505,16 @@ func extractQuotedStrings(s string) ([]string, error) {
 	}
 	return found, nil
 }
+
+// quoteIdentifier quotes an "identifier" (e.g. a table or a column name) to be used as part of an SQL statement.
+//
+// This is a copy of the same function as found in lib/pq (https://github.com/lib/pq/blob/v1.10.4/conn.go#L1640)
+// and was added as part of our pq -> pgx migration. The pgx package doesn't expose a similar function at this
+// time (https://github.com/jackc/pgx/issues/868).
+func quoteIdentifier(name string) string {
```

Comment:
is there a reason that we define this for each db instead of in a shared library?

Reply:
No particular reason other than it being short enough for it to be copied vs pulling it in as a dependency (in the spirit of the [Go proverb](https://go-proverbs.github.io/) around copying). However, since we'll be attaching the copyright and license into the file, it might be worth pulling this out into its own helper package along with `parseURL`.

Reply:
Moved both helpers to the sdk package and attached the copyright and license in [cc7abda](https://github.com/hashicorp/vault/pull/15343/commits/cc7abda403d06f29677e9ad9556c0da070048005).